.github/workflows/prom_exporter_ext.yml

```diff
@@ -11,30 +11,81 @@ on:
       - main
     paths:
     - '.github/workflows/postgres_extension.yml'
-    - 'extensions/**'
+    - 'extensions/prometheus_exporter/**'
   # Always run tests on main or release branches
   push:
     branches:
       - main
       - release**
 
-
 jobs:
-  build_and_test:
-    name: Rust project
+  lint:
+    name: Run linters
     runs-on: ubuntu-20.04
     steps:
       - uses: actions/checkout@v2
+      - name: Install Rust minimal nightly with clippy and rustfmt
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          components: rustfmt, clippy
+      - uses: Swatinem/rust-cache@v2
+        with:
+          prefix-key: "prom-extension-lint"
+          workspaces: |
+            extensions/prometheus_exporter
+          # Additional directories to cache
+          cache-directories: |
+            /home/runner/.pgx
+      - uses: ./.github/actions/pgx-init
+        with:
+          working-directory: extensions/prometheus_exporter
+      - name: Cargo format
+        run: cargo +nightly fmt --check
+      - name: Clippy
+        run: cargo clippy
 
-      - uses: actions-rs/toolchain@v1
+  test:
+    name: Run tests
+    runs-on: ubuntu-20.04
+    strategy:
+      # fail-fast means to cancel all jobs if one fails
+      fail-fast: false
+      matrix:
+        pg_version:
+          - 14
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Rust stable toolchain
+        uses: actions-rs/toolchain@v1
         with:
           toolchain: stable
-
-      - name: setup pgx
-        run: cargo install --locked cargo-pgx
-
-      - name: init
-        run: cargo pgx init --pg14 download
-
+      - uses: Swatinem/rust-cache@v2
+        with:
+          prefix-key: "prom-extension-test-${{ matrix.pg_version }}"
+          workspaces: |
+            extensions/prometheus_exporter
+          # Additional directories to cache
+          cache-directories: |
+            /home/runner/.pgx
+      - uses: ./.github/actions/pgx-init
+        with:
+          working-directory: extensions/prometheus_exporter
+          pg_version: ${{ matrix.pg_version }}
       - name: test
-        run: cargo test --no-default-features --features pg14
+        run: |
+          set -xe
+          # Debugging information
+          ls -ltrah /home/runner/.pgx/
+          ls -ltrah /home/runner/.pgx/data-${{ matrix.pg_version }}/
+          whoami
+          cat /home/runner/.pgx/data-${{ matrix.pg_version }}/postgresql.conf
+          cat /home/runner/.pgx/data-${{ matrix.pg_version }}/postgresql.auto.conf
+          ls ./target/target/pgx-test-data-${{ matrix.pg_version }}
+          # ensure test data directory is deleted before starting tests
+          rm -rf ./target/target/pgx-test-data-${{ matrix.pg_version }}
```

Comment:
I'm having some issue with failure in pg_test to modify the config file. I found log from working case that the data directory is here:
```
target/pgx-test-data-14
```

so, we can try to delete it before running the test, this might help. The working logs include creating the directory:

```
The files belonging to this database system will be owned by user "runner".
This user must also own the server process.

The database cluster will be initialized with locale "C.UTF-8".
The default database encoding has accordingly been set to "UTF8".
The default text search configuration will be set to "english".

Data page checksums are disabled.

creating directory /home/runner/work/coredb/coredb/extensions/prometheus_exporter/target/pgx-test-data-14 ... ok
creating subdirectories ... ok
selecting dynamic shared memory implementation ... posix
selecting default max_connections ... 100
selecting default shared_buffers ... 128MB
selecting default time zone ... Etc/UTC
creating configuration files ... ok
running bootstrap script ... ok
performing post-bootstrap initialization ... ok
initdb: warning: enabling "trust" authentication for local connections
syncing data to disk ... ok

You can change this by editing pg_hba.conf or using the option -A, or
--auth-local and --auth-host, the next time you run initdb.

Success. You can now start the database server using:

    /home/runner/.pgx/14.6/pgx-install/bin/pg_ctl -D /home/runner/work/coredb/coredb/extensions/prometheus_exporter/target/pgx-test-data-14 -l logfile start

test webserver::tests::pg_test_pg_uptime ... ok
```

Reply:
ok I think that worked